pinot-segment-local/src/main/java/org/apache/pinot/segment/local/indexsegment/immutable/ImmutableSegmentImpl.java

```diff
@@ -200,6 +200,74 @@ private File getValidDocIdsSnapshotFile() {
         V1Constants.VALID_DOC_IDS_SNAPSHOT_FILE_NAME);
   }
 
+  @Nullable
+  public MutableRoaringBitmap loadQueryableDocIdsFromSnapshot() {
+    File queryablDocIdsSnapshotFile = getQueryableDocIdsSnapshotFile();
+    if (queryablDocIdsSnapshotFile.exists()) {
+      try {
+        byte[] bytes = FileUtils.readFileToByteArray(queryablDocIdsSnapshotFile);
+        MutableRoaringBitmap queryableDocIds =
+            new ImmutableRoaringBitmap(ByteBuffer.wrap(bytes)).toMutableRoaringBitmap();
+        LOGGER.info("Loaded queryableDocIds for segment: {} with: {} queryable docs", getSegmentName(),
+            queryableDocIds.getCardinality());
+        return queryableDocIds;
+      } catch (Exception e) {
+        LOGGER.warn("Caught exception while loading queryableDocIds from snapshot file: {}, ignoring the snapshot",
+            queryablDocIdsSnapshotFile);
+      }
+    }
+    return null;
+  }
+
+  public void persistQueryableDocIdsSnapshot() {
```

Comment:
since a lot of code is shared with valid docs ids , can we create a method to take Snapshotfile as an argument and call that from persistQueryableDocIdsSnapshot and persistValidDocIdsSnapshot ? 

its applicable for loadQueryableDocIdsFromSnapshot too .